editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java

```diff
@@ -0,0 +1,350 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.esri.samples.edit_features_with_feature_linked_annotation;
+
+import java.io.File;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import javafx.application.Application;
+import javafx.geometry.Point2D;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.input.MouseButton;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.data.Feature;
+import com.esri.arcgisruntime.data.Geodatabase;
+import com.esri.arcgisruntime.geometry.GeometryEngine;
+import com.esri.arcgisruntime.geometry.GeometryType;
+import com.esri.arcgisruntime.geometry.Part;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.geometry.Polyline;
+import com.esri.arcgisruntime.geometry.PolylineBuilder;
+import com.esri.arcgisruntime.geometry.ProximityResult;
+import com.esri.arcgisruntime.layers.AnnotationLayer;
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.IdentifyLayerResult;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+public class EditFeaturesWithFeatureLinkedAnnotationSample extends Application {
+
+  private MapView mapView;
+  private FeatureLayer addressPointFeatureLayer;
+  private FeatureLayer parcelLinesFeatureLayer;
+  private AnnotationLayer addressPointsAnnotationLayer;
+  private AnnotationLayer parcelLinesAnnotationLayer;
+  private Feature selectedFeature = null;
+  private boolean selectedFeatureIsPolyline = false;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add JavaFX scene to stage
+      stage.setTitle("Edit Features With Feature-linked Annotation Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create the map with a light gray canvas basemap centered on Loudoun, Virginia
+      ArcGISMap map = new ArcGISMap(Basemap.Type.LIGHT_GRAY_CANVAS_VECTOR, 39.0204, -77.4159, 18);
+
+      // create a map view and set its map
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // add the map view to stack pane
+      stackPane.getChildren().addAll(mapView);
+      
+      // create and load the geodatabase
+      File geodatabaseFile = new File(System.getProperty("data.dir"),
+        "./samples-data/loudon/loudoun_anno.geodatabase");
+      Geodatabase geodatabase = new Geodatabase(geodatabaseFile.getAbsolutePath());
```

Comment:
we should make this an instance variable. Since it is a loadable, and we are waiting on it to finish loading, there is a small chance it's initial reference will be garbage collected and by the time the addDoneLoadingListener has returned, the initial `geodatabase` reference is lost and the app will hang.  Let's add the following up above the start method:

```java
  private Geodatabase geodatabase; // keep loadable in scope to avoid garbage collection

```
```suggestion
      geodatabase = new Geodatabase(geodatabaseFile.getAbsolutePath());
```